### PR TITLE
[Numpy] FFI: max/min/amax/amin

### PR DESCRIPTION
**Pull request by Tommliu**
## Description ##
New FFI imperative invocation for max, min ,amax, amin

| Numpy Operator   |   Old FFI (ctypes) (us) |   New FFI (cython) (us) |
|------------------|-------------------------|-------------------------|
| amax             |                   26.98 |                    7.94 |
| amin             |                   26.9  |                    7.84 |
| max              |                   27.51 |                    7.87 |
| min              |                   27.02 |                    7.92 |
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/numpy/multiarray.py [hunk 1]
```diff
@@ -6450,6 +6578,132 @@ def argmin(a, axis=None, out=None):
...
+    array([0., 1.])
+    >>> np.min(a, axis=1)   # Minima along the second axis
+    array([0., 2.])
+    >>> b = np.arange(5, dtype=np.float32)
+    >>> b[2] = np.nan
+    >>> np.min(b)
+    array(0.) # nan will be ignored
+    """
+    return _mx_nd_np.amin(a, axis=axis, out=out, keepdims=keepdims)
+
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: one more blank line below.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 2]
```diff
@@ -581,7 +582,7 @@ def argmin(self, axis=None, out=None):  # pylint: disable=arguments-differ
         Refer to `mxnet.numpy.argmax` for full documentation."""
         return argmin(self, axis, out)
 
-    def clip(self, min=None, max=None, out=None):  # pylint: disable=arguments-differ
+    def clip(self, min=None, max=None, out=None):  # pylint: disable=arguments-differ,redefined-outer-name
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: add a space before `redefined-outer-name`

**Comment by Tommliu** [comment]
@mxnet-bot run ci [unix-gpu, windows-gpu]

**Comment by Tommliu** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by Tommliu** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by Tommliu** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by Tommliu** [comment]
@mxnet-bot run ci [windows-gpu]

**Comment by haojin2** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by haojin2** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by haojin2** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by Tommliu** [comment]
@mxnet-bot run ci [unix-gpu, unix-cpu, centos-cpu, website, windows-cpu, sanity]

**Comment by Tommliu** [comment]
@haojin2 last FFI Pr to be merged

**Comment by Tommliu** [comment]
@haojin2

**Pull request merged**